### PR TITLE
Real-space cutoffs

### PR DESCRIPTION
**Pull request by marvinfriede**
Real-space cutoffs for the coordination number and the dispersion energy are implemented to (finally) achieve full consistency with the total dispersion energy of the Fortran implementation.

### CODE REVIEW

**Review by awvwgk — CHANGES_REQUESTED** [review]
Let's provide all cutoff values in the constructor rather than having to construct an incomplete object and adding the attributes with setters.

Also, no C-style casting in C++.

**Comment by marvinfriede** [comment]
> Let's provide all cutoff values in the constructor rather than having to construct an incomplete object and adding the attributes with setters.

I don't understand that part. The constructor writes the default values to all members [here](https://github.com/dftd4/cpp-d4/blob/a705cfd1d3c57447801e63e98a664e8d56bd6a19/src/dftd_cutoff.cpp#L35). The setters are just for changing it (required within a test).
 
> Also, no C-style casting in C++.

So just multiplying by 1.0?

**Comment by awvwgk** [comment]
> I don't understand that part. The constructor writes the default values to all members [here](https://github.com/dftd4/cpp-d4/blob/a705cfd1d3c57447801e63e98a664e8d56bd6a19/src/dftd_cutoff.cpp#L35). The setters are just for changing it (required within a test).

Why not have the tests recreate the object with the correct values in the constructor then? Or directly setting the value of the members as they are public.

> So just multiplying by 1.0?

https://en.cppreference.com/w/cpp/language/static_cast

**Comment by marvinfriede** [comment]
> > I don't understand that part. The constructor writes the default values to all members [here](https://github.com/dftd4/cpp-d4/blob/a705cfd1d3c57447801e63e98a664e8d56bd6a19/src/dftd_cutoff.cpp#L35). The setters are just for changing it (required within a test).
> 
> Why not have the tests recreate the object with the correct values in the constructor then? Or directly setting the value of the members as they are public.

Oh. I forgot about the members being public...

File: src/dftd_cutoff.cpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  cn_eeq = cn_eeq_default;
+};
+
+void TCutoff::disable() {
+  const double new_cutoff = 999999999;
+  disp2 = new_cutoff;
+  disp3 = new_cutoff;
+  cn = new_cutoff;
+  cn_eeq = new_cutoff;
+};
```
Comment (awvwgk) [inline src/dftd_cutoff.cpp:48]: Why do we need this function? Also, why the magic number rather than using numeric limits?

Comment (marvinfriede) [reply src/dftd_cutoff.cpp:48]: I wrote some tests before implementing the cutoffs. So now it is just a double check to see if the energy is correct with and without using cutoffs. Should I remove the no-cutoff test?

**Comment by marvinfriede** [comment]
Can I merge this?

**Review by awvwgk — APPROVED** [review]

**Pull request merged**